plugins/node/instrumentation-kafkajs/src/instrumentation.ts

```diff
@@ -111,11 +229,39 @@ export class KafkaJsInstrumentation extends InstrumentationBase<KafkaJsInstrumen
           instrumentation._getConsumerRunPatch()
         );
 
+        instrumentation._setConsumerEventListeners(newConsumer);
+
         return newConsumer;
       };
     };
   }
 
+  private _setConsumerEventListeners(consumer: ConsumerExtended) {
+    if (consumer[EVENT_LISTENERS_SET]) return;
+
+    consumer.on(consumer.events.END_BATCH_PROCESS, event =>
+      this._consumedMessages.add(event.payload.batchSize, {
+        [ATTR_MESSAGING_SYSTEM]: MESSAGING_SYSTEM_VALUE_KAFKA,
+        [ATTR_MESSAGING_OPERATION_NAME]: 'receive',
+        [ATTR_MESSAGING_DESTINATION_NAME]: event.payload.topic,
+        [ATTR_MESSAGING_DESTINATION_PARTITION_ID]: String(
+          event.payload.partition
+        ),
+      })
+    );
+    consumer.on(consumer.events.REQUEST, event => {
+      const [address, port] = event.payload.broker.split(':');
+      this._clientDuration.record(event.payload.duration / 1000, {
+        [ATTR_MESSAGING_SYSTEM]: MESSAGING_SYSTEM_VALUE_KAFKA,
+        [ATTR_MESSAGING_OPERATION_NAME]: `${event.payload.apiName}`, // potentially suffix with @${event.payload.apiVersion}?
+        [ATTR_SERVER_ADDRESS]: address,
+        [ATTR_SERVER_PORT]: Number.parseInt(port, 10),
+      });
+    });
+
+    consumer[EVENT_LISTENERS_SET] = true;
+  }
+
```

Comment:
I tried for quite awhile to figure out a way to wrap/stub/spy/etc on these (as well as the producer event listeners below) and failed. Very open to suggestions.